packages/apps/shopify-app-remix/src/server/shopify-app.ts

```diff
@@ -71,12 +72,22 @@ export function shopifyApp<
   }
 
   const params: BasicParams = {api, config, logger};
+
+  let strategy;
+  if (config.distribution === AppDistribution.ShopifyAdmin) {
+    strategy = new MerchantCustomAuth(params);
+  } else if (
+    config.future.unstable_newEmbeddedAuthStrategy &&
+    config.isEmbeddedApp
+  ) {
+    strategy = new TokenExchangeStrategy(params);
```

Comment:
I'm surprised making the storage optional didn't require any changes in the pre-existing strategies - is that type incorrect in those?